packages/muelu/test/unit_tests/RebalanceAcFactory.cpp

```diff
@@ -0,0 +1,470 @@
+
+//    RebalancedAFact->AddRebalanceFactory(RebalancedPFact);
+//    RebalancedAFact->AddRebalanceFactory(RebalancedRFact);
+
+
+    M.SetFactory("A", RebalancedAFact);
+    M.SetFactory("P", RebalancedPFact);
+    M.SetFactory("R", RebalancedRFact);
+    M.SetFactory("Nullspace",   RebalancedPFact);
+//    M.SetFactory("Coordinates", RebalancedPFact);
+    M.SetFactory("Importer",    RepartitionFact);
+
+    int startLevel = 0;
+    H->Setup(M, startLevel, optMaxLevels);
+
+    RebalancedAFact->Build(*level, *level);
+
+
+  }
+  TEUCHOS_UNIT_TEST_TEMPLATE_4_DECL(RebalanceAcFactory, Build2, Scalar, LocalOrdinal, GlobalOrdinal, Node)
+  {
+    #include <MueLu_UseShortNames.hpp>
+    MUELU_TESTING_SET_OSTREAM;
+    MUELU_TESTING_LIMIT_SCOPE(Scalar,GlobalOrdinal,Node);
+    out << "version: " << MueLu::Version() << std::endl;
+
+    // - Smoothed-Aggregation
+    Scalar optSaDamping = 4./3;
+      int optMaxCoarseSize = 50;
+      LO optMinRowsPerProc = 2000;
+      double optNnzImbalance = 1.2;
+      Teuchos::CommandLineProcessor clp(false);
+LO  optMaxLevels     = 10;
+      // Default is Laplace1D with nx = 8748.
+      // It's a nice size for 1D and perfect aggregation. (6561 = 3^8)
+      //Nice size for 1D and perfect aggregation on small numbers of processors. (8748 = 4*3^7)
+      Galeri::Xpetra::Parameters<GO> matrixParameters(clp, 8748); // manage parameters of the test case
+      Xpetra::Parameters xpetraParameters(clp);                   // manage parameters of xpetra
+
+      // create coarsest smoother
+      RCP<SmootherPrototype> coarsestSmooProto;
+      std::string type = "";
+      Teuchos::ParameterList coarsestSmooList;
+  #if defined(HAVE_AMESOS_SUPERLU)
+      coarsestSmooProto = Teuchos::rcp( new DirectSolver("Superlu", coarsestSmooList) );
+  #else
+      coarsestSmooProto = Teuchos::rcp( new DirectSolver("Klu", coarsestSmooList) );
+  #endif
+      RCP<SmootherFactory> coarsestSmooFact = rcp(new SmootherFactory(coarsestSmooProto, Teuchos::null));
+
+
+      RCP<const Map> map;
+      RCP<Matrix> A;
+
+      RCP<MultiVector> coordinates;
+
+      // USER GUIDE // define communicator
+      RCP<const Teuchos::Comm<int> > comm = Teuchos::DefaultComm<int>::getComm();
+      Xpetra::UnderlyingLib lib = xpetraParameters.GetLib();
+
+      map = MapFactory::Build(lib, matrixParameters.GetNumGlobalElements(), 0, comm);
+      Teuchos::RCP<Galeri::Xpetra::Problem<Map,CrsMatrixWrap,MultiVector> > Pr =
+          Galeri::Xpetra::BuildProblem<SC,LO,GO,Map,CrsMatrixWrap,MultiVector>(matrixParameters.GetMatrixType(), map, matrixParameters.GetParameterList()); //TODO: Matrix vs. CrsMatrixWrap
+      A = Pr->BuildMatrix();
+      if (matrixParameters.GetMatrixType() == "Laplace1D") {
+        coordinates = Galeri::Xpetra::Utils::CreateCartesianCoordinates<SC, LO, GO, Map, MultiVector>("1D", map, matrixParameters.GetParameterList());
+      }
+      else if (matrixParameters.GetMatrixType() == "Laplace2D") {
```

Comment:
The default matrixType is Laplace1D.  The unit test will never produce a Laplace2D or Laplac3D matrix, so I think these can be deleted.